sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -79,247 +99,318 @@ function getProvider(props: any): dynamic.ResourceProvider {
 // This allows the creation of the replacement resource to use the new provider while the deletion of the old
 // resource uses the provider with which it was created.
 
-function cancelRPC(call: any, callback: any): void {
-    callback(undefined, new emptyproto.Empty());
-}
+class ResourceProviderService implements provrpc.IResourceProviderServer {
+    [method: string]: grpc.UntypedHandleCall;
 
-function configureRPC(call: any, callback: any): void {
-    const resp = new provproto.ConfigureResponse();
-    resp.setAcceptsecrets(false);
-    callback(undefined, resp);
-}
-
-async function invokeRPC(call: any, callback: any): Promise<void> {
-    const req: any = call.request;
+    cancel(call: any, callback: any): void {
+        callback(undefined, new emptyproto.Empty());
+    }
 
-    // TODO[pulumi/pulumi#406]: implement this.
-    callback(new Error(`unknown function ${req.getTok()}`), undefined);
-}
+    async configure(
```

Comment:
The diff here is unfortunately very noisy, what's changed is the implementation of `configure` here, and adding stubs for the non-implemented methods like `attach`, `call`, ... below.